File: common/src/main/java/generations/gg/generations/core/generationscore/common/world/level/block/generic/GenericModelMultiBlock.java
package generations.gg.generations.core.generationscore.common.world.level.block.generic;

import net.minecraft.world.level.block.Block;
import net.minecraft.world.level.block.HorizontalDirectionalBlock;
import net.minecraft.world.level.block.Mirror;
import net.minecraft.world.level.block.Rotation;
import net.minecraft.world.level.block.state.BlockState;
import net.minecraft.world.level.block.state.StateDefinition;
import net.minecraft.world.level.block.state.properties.DirectionProperty;
import org.jetbrains.annotations.NotNull;

@SuppressWarnings({"deprecation"})
public class GenericModelMultiBlock extends Block {

    // Work in progress, unsure what exactly I need to do here yet

    public static final DirectionProperty FACING = HorizontalDirectionalBlock.FACING;

    public GenericModelMultiBlock(Properties properties, int width, double height, int length) {
        super(properties);
    }

    @Override
    protected void createBlockStateDefinition(StateDefinition.Builder<Block, BlockState> builder) {
        builder.add(FACING);
    }

    @Override
    public @NotNull BlockState rotate(BlockState state, Rotation rotation) {
        return state.setValue(FACING, rotation.rotate(state.getValue(FACING)));
    }

    @Override
    public @NotNull BlockState mirror(@NotNull BlockState state, @NotNull Mirror mirror) {
        return mirror == Mirror.NONE ? state : state.rotate(mirror.getRotation(state.getValue(FACING)));
    }



}

File: common/src/main/java/generations/gg/generations/core/generationscore/common/world/level/block/generic/GenericModelMultiBlock.kt
package generations.gg.generations.core.generationscore.common.world.level.block.generic

import net.minecraft.world.level.block.Block
import net.minecraft.world.level.block.HorizontalDirectionalBlock
import net.minecraft.world.level.block.Mirror
import net.minecraft.world.level.block.Rotation
import net.minecraft.world.level.block.state.BlockState
import net.minecraft.world.level.block.state.StateDefinition
import net.minecraft.world.level.block.state.properties.DirectionProperty

@Suppress("deprecation")
class GenericModelMultiBlock(properties: Properties, width: Int, height: Double, length: Int) :
    Block(properties) {
    override fun createBlockStateDefinition(builder: StateDefinition.Builder<Block, BlockState>) {
        builder.add(FACING)
    }

    public override fun rotate(state: BlockState, rotation: Rotation): BlockState {
        return state.setValue(FACING, rotation.rotate(state.getValue(FACING)))
    }

    public override fun mirror(state: BlockState, mirror: Mirror): BlockState {
        return if (mirror == Mirror.NONE) state else state.rotate(
            mirror.getRotation(
                state.getValue(
                    FACING
                )
            )
        )
    }


    companion object {
        // Work in progress, unsure what exactly I need to do here yet
        val FACING: DirectionProperty = HorizontalDirectionalBlock.FACING
    }
}
